### ISSUE TITLE
Discussion: default n_jobs for Featurizers

### ISSUE THREAD

**Issue by computron** [hackingmaterials/matminer#162]
Currently the default n_jobs for featurizers is 1, i.e., serial jobs.:

matminer/featurizers/base.py:105

This choice is in-line with what sklearn featurizers do, e.g.:

http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html

I'd suggest being more aggressive and setting this equal to cpu_count() - i.e., to use full parallelization by default. For example, the multiprocessing module uses cpu_count when the number of processes is set as ``None`` (default is full parallelization).

Regardless of the default, I can see use cases for both and it would be nice to have a "set-it-once-and-forget-it" way to configure n_jobs. e.g., a config variable one can set at the beginning of a Jupyter notebook.

**Comment by ardunn** [comment]
My vote is for `cpu_count()` as default

**Comment by WardLT** [comment]
I vote `cpu_count()` too. I was originally feeling conservative, but don't think we'll have too many cases where someone would be upset our code is running in parallel.

I also like the idea of adding a configuration variable.
